src/main/java/github/scarsz/discordsrv/DiscordSRV.java

```diff
@@ -721,19 +726,42 @@ private List<InetAddress> lookupPublic(String host) throws UnknownHostException
         dispatcher.setMaxRequestsPerHost(20); // most requests are to discord.com
         ConnectionPool connectionPool = new ConnectionPool(5, 10, TimeUnit.SECONDS);
 
-        OkHttpClient httpClient = new OkHttpClient.Builder()
+        String proxyHost = config.getString("ProxyHost").trim();
+        String proxyPort = config.getString("ProxyPort").trim();
+        final String authUser = config.getString("ProxyUser").trim();
+        final String authPassword = config.getString("ProxyPassword").trim();
+
+        OkHttpClient httpClient; // Initialized below
+        OkHttpClient.Builder httpClientBuilder = new OkHttpClient.Builder()
                 .dispatcher(dispatcher)
                 .connectionPool(connectionPool)
                 .dns(dns)
-                // more lenient timeouts (normally 10 seconds for these 3)
                 .connectTimeout(20, TimeUnit.SECONDS)
                 .readTimeout(20, TimeUnit.SECONDS)
                 .writeTimeout(20, TimeUnit.SECONDS)
                 .hostnameVerifier(noopHostnameVerifier.isPresent() && noopHostnameVerifier.get()
                         ? (hostname, sslSession) -> true
-                        : OkHostnameVerifier.INSTANCE
-                )
-                .build();
+                        : OkHostnameVerifier.INSTANCE);
+
+        // Reduce proxy username and password logic duplication
+        if (proxyHost != null && !proxyHost.isEmpty()) {
```

Comment:
`proxyHost` is always a `String`. Add a check against the default value of `https://example.com` instead.